lib/answer_composition/pipeline/bedrock_structured_answer_composer.rb

```diff
@@ -0,0 +1,99 @@
+module AnswerComposition::Pipeline
+  class BedrockStructuredAnswerComposer
+    BEDROCK_MODEL = "eu.anthropic.claude-3-5-sonnet-20240620-v1:0".freeze
+
+    def self.call(...) = new(...).call
+
+    def initialize(context)
+      @context = context
+    end
+
+    def call
+      start_time = Clock.monotonic_time
+
+      response = bedrock_client.converse(
+        system: [{ text: system_prompt }],
+        model_id: BEDROCK_MODEL,
+        messages:,
+        inference_config:,
+        tool_config:,
+      )
+
+      context.answer.assign_llm_response("structured_answer", response.to_h)
+      message = response["output"]["message"]["content"][0]["tool_use"]["input"]["answer"]
```

Comment:
```suggestion
      message = response.dig("output", "message", "content", 0, "tool_use", "input", "answer")
```

using dig should be a bit cleaner. 

So we don't need to do a JSON parse, it merges the JSON into the response rather than returning a string of JSON?

Reply:
Yes, the client returns the response as nested structs.